qr-code/src/main/java/com/adyen/checkout/qrcode/DefaultQRCodeDelegate.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 15/8/2022.
+ */
+
+package com.adyen.checkout.qrcode
+
+import androidx.annotation.VisibleForTesting
+import com.adyen.checkout.components.model.payments.response.Action
+import com.adyen.checkout.components.model.payments.response.QrCodeAction
+import com.adyen.checkout.components.status.DefaultStatusRepository
+import com.adyen.checkout.components.status.StatusRepository
+import com.adyen.checkout.components.status.api.StatusResponseUtils
+import com.adyen.checkout.components.status.model.StatusResponse
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.exception.ComponentException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.concurrent.TimeUnit
+
+@Suppress("TooManyFunctions")
+internal class DefaultQRCodeDelegate(
+    private val statusRepository: StatusRepository,
+    private val statusCountDownTimer: QRCodeCountDownTimer,
+) : QRCodeDelegate {
+
+    private val _outputDataFlow = MutableStateFlow<QRCodeOutputData?>(null)
+    override val outputDataFlow: Flow<QRCodeOutputData?> = _outputDataFlow
+
+    override val outputData: QRCodeOutputData? get() = _outputDataFlow.value
+
+    private val _exceptionFlow = MutableSharedFlow<CheckoutException>(0, 1, BufferOverflow.DROP_OLDEST)
+    override val exceptionFlow: Flow<CheckoutException> = _exceptionFlow
```

Comment:
This flow is not observed in the component right?

Reply:
Very sharp! Apparently, I also missed the details flow 👀

Reply:
Minor detail 😂